### ISSUE TITLE
[FEAT] Classifying questions according to difficulty

### ISSUE THREAD

**Issue by rahulsagu** [Preclinify/Tickets#28]
**As a** user, I want to use user metrics to determine the difficulty of the question so that users can test themselves according to the level of their confidence.

This would be determined after the launch where a minimum threshold of X students attempt the question. The difficulty can be decided based on the percentage of correct first attempts of a question. For instance:
- A question can be deemed 'easy' difficulty if approximately 66% of the user base (or above) correctly answers the question. 
- A question can be deemed 'medium' difficulty if between approximately 33-66% of the user base correctly answers the question. 
- A question can be deemed 'hard' difficulty if approximately 33% of the user base (or less) correctly answers the question. 

I would want this feature to appear on the page where students can select the topics they want to answer questions on - this way, they can select the difficulty of the questions before commencing a session. This would appear in the modal where students can choose between Exam Mode and Endless Mode. 

In terms of a logo:
- Easy difficulty can be represented with one wing.
- Medium difficulty can be represented with two wings.
- Hard difficulty can be represented with three wings.

**Comment by Joecey** [comment]
Making me hungry with all this wing talk....

- But for real, when a user selects a "medium" difficulty, does this include easy and medium questions? Or should it exclusively show "medium" questions. 
- Should there be a threshold for the amount of times a question needs to be answered before indicating its difficulty level. A better question would be, if we created a brand new question --> what is the default behaviour of this?

**Comment by rahulsagu** [comment]
Hahahaha it's your sign to go wingstop rn :o

- So if a user was to select "medium" difficulty, this would be exclusively medium. To have easy and medium questions, they would select both prior to commencing a session,
- Now that is a good question. At the moment, the questions will remain unclassified. I spoke to Rohan and we think once the current questions have been attempted during the beta launch, we can classify them accordingly. This could perhaps be a feature implemented in the open launch as opposed to the closed launch.

**Comment by Joecey** [comment]
we could potentially add a counter to see how many times a question has been answered? Adding the amount of times it has been answered correctly could be done too? **This would purely used just for us for post-mortem, and not be shown on the site at all** 

What do you think, I would ask of you to create another ticket if you'd like the idea because this would be data mutation rather than data fetching (hope that makes sense :3)

**Comment by rahulsagu** [comment]
Yeah that sounds good having that viewable to just the back end and not the user end. I'm happy to create another ticket for sure!! In fact, I'll make two:
- Adding the counter
- Also showing the percentage of students selecting each option (A-E) within an MCQ - this will be viewable to the user end too to and will help us when classifying the difficulty of questions
